### ISSUE TITLE
Continuing training resets logger step

### ISSUE THREAD

**Issue by wittenator** [PyTorchLightning/pytorch-lightning#6435]
## 🐛 Bug
I am running Pytorch Lightning in a federated learning setting. Therefore I have several models and I need to instantiate a Trainer object for one model multiple times. Every time I do that the associated logger resets the epoch and logs the metrics on top of each other in the plots. Since instantiating a new Trainer object to continue training a model is allowed as far as I know: Do you know if that is expected behaviour and whether there is a workaround?

In this picture is an example of the logging output of three consecutively called trainers with a common logger.
![image](https://user-images.githubusercontent.com/9154515/110460909-1727c500-80cf-11eb-9ac3-d119bb269a2f.png)

## Please reproduce using the BoringModel

The Colab currently does not work with lightning and lightning_bolts.

### To Reproduce

Create a TestTubeLogger and instantiate multiple Trainers with a common logger for the same model and fit the trainers consecutively.

### Expected behaviour
The x-axis of the logged metrics should not reset between runs.

### Environment
* CUDA:
	- GPU:
	- available:         False
	- version:           None
* Packages:
	- numpy:             1.20.1
	- pyTorch_debug:     False
	- pyTorch_version:   1.8.0+cpu
	- pytorch-lightning: 1.2.1
	- tqdm:              4.58.0
* System:
	- OS:                Linux
	- architecture:
		- 64bit
		- ELF
	- processor:         x86_64
	- python:            3.8.5
	- version:           #74-Ubuntu SMP Wed Jan 27 22:54:38 UTC 2021

**Comment by wittenator** [comment]
Hi @tchaton ,

thanks for the tip with the reloading! I didn't try that yet because saving and reading is a bit of overhead. I would look into adding persistent step/epoch counting. As far as I see it, that should be a pretty small addition since there must be some code to count steps and epochs in order to display that in the logger for one trainer. I would only need to persist and accumulate that data in the model. Do you have some pointers into which class I should look?
